framework/src/problems/EigenProblem.C

```diff
@@ -143,6 +144,57 @@ EigenProblem::computeJacobianTag(const NumericVector<Number> & soln,
   _nl_eigen->disassociateMatrixFromTag(jacobian, tag);
 }
 
+void
+EigenProblem::computeMatricesTags(
+    const NumericVector<Number> & soln,
+    const std::vector<std::unique_ptr<SparseMatrix<Number>>> & jacobians,
+    const std::set<TagID> & tags)
+{
+  TIME_SECTION(_compute_jacobian_tag_timer);
+
+  if (jacobians.size() != tags.size())
+    mooseError("The number of matrices ",
+               jacobians.size(),
+               " does not equal the number of tags ",
+               tags.size());
+
+  _fe_matrix_tags.clear();
+
+  _nl_eigen->setSolution(soln);
+
+  _nl_eigen->disassociateAllTaggedMatrices();
+
+  unsigned int i = 0;
+  for (auto tag : tags)
+    _nl_eigen->associateMatrixToTag(*(jacobians[i++]), tag);
+
+  computeJacobianTags(tags);
+
+  i = 0;
+  for (auto tag : tags)
+    _nl_eigen->disassociateMatrixFromTag(*(jacobians[i++]), tag);
+}
+
+void
+EigenProblem::computeJacobianBlocks(std::vector<JacobianBlock *> & blocks)
+{
+  TIME_SECTION(_compute_jacobian_blocks_timer);
+
+  if (_displaced_problem)
+  {
+    _aux->compute(EXEC_PRE_DISPLACE);
+    _displaced_problem->updateMesh();
+  }
+
+  _aux->compute(EXEC_NONLINEAR);
+
+  _currently_computing_jacobian = true;
+
+  _nl->computeJacobianBlocks(blocks, {_nl_eigen->precondMatrixTag()});
+
+  _currently_computing_jacobian = false;
```

Comment:
Who calls this? Are we sure we're done computing the Jacobian once this method returns?

Reply:
It is called by a preconditioner. I may not need this flag at all

Reply:
OK, this code was copied from FEProblemBase. The motivation is to pass the preconditioning tag so the matrix can be assembled properly for eigenvalue problems.

I would like to keep the code consistent with the FEProblemBase. At this point, I not sure why we call `meshUpdate` , but I would prefer to believe the original author has a good reason for that.

Reply:
The same for `_currently_computing_jacobian`